ui: jobs redesign follow up

vladlos opened the pull request:
various design updates for jobs page

Resolves: #45259

Release note (ui): none

vladlos committed:
ui: Job list

Updated status styling

Release note (ui): none;

vladlos committed:
ui: Jobs filter

Updated statuses in filter

Release note (ui): none

vladlos committed:
ui: Job list

updated empty and no result state

Release note (ui): none

vladlos committed:
ui: Font sizes on job details page

Updated font sizes on page

Release not (ui): none

vladlos committed:
ui: Job details creation time

Updated creation date format

Release note (ui): none

vladlos committed:
ui: Job details page status

Updated status displays for succeeded failed pending canceled paused to new pill style

Release note (ui): none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45386)
<!-- Reviewable:end -->

vladlos committed:
ui: Filter caret

removed caret up (always point down)

Release note (ui): none

koorosh reviewed (changes requested):
Suggestions for Empty component:
- Empty component has to expose props: `title`, `body`, and `footer`. It will allow to define whatever you
want in footer (Button, or Link).
- Use `src/components/button` instead of Button from Ant.design. It is already styled according to new designs.

vladlos committed:
ui: Job list

imported own components, refactoring

Release note (ui): none

dhartunian commented on pkg/ui/src/views/statements/statementsPage.tsx:
```diff
@@ -158,6 +158,13 @@ export class StatementsPage extends React.Component<StatementsPageProps & RouteC
+  noStatementsResult = () => (
+    <>
+      <p>There are no SQL statements that match your search or filter since this page was last cleared.</p>
+      <a href="https://www.cockroachlabs.com/docs/stable/admin-ui-statements-page.html" target="_blank">Learn more about the statement page</a>
```
pluralize statement(s)

dhartunian commented on pkg/ui/src/views/jobs/jobStatusOptions.ts:
```diff
@@ -29,13 +28,40 @@ export function jobHasOneOfStatuses(job: Job, ...statuses: string[]) {
+      };
+    case JOB_STATUS_RUNNING:
+      return {
+        label: "CDC Running",
```
I'm realizing there might be some confusion around "running". The running status is actually different from "CDC running". CDC refers to a CHANGEFEED job which itself can be in a running state.

I will ask Anne to clarify when "CDC Running" should be shown.

dhartunian commented on pkg/ui/src/views/jobs/duration.tsx:
```diff
@@ -30,8 +29,8 @@ export class Duration extends React.PureComponent<{ job: Job }> {
-    if (jobHasOneOfStatuses(this.props.job, JOB_STATUS_PENDING, JOB_STATUS_PAUSED)) {
-      return _.capitalize(this.props.job.status);
+    if (jobHasOneOfStatuses(this.props.job, JOB_STATUS_PENDING)) {
+      return "Waiting for GG TCL";
```
The message in the design was there as an example not as a permanent status for all pending jobs. The actual message in the code is `Waiting for GC TTL` and is set on the job as its "running_status" field.

I think for now, if the job has a running status, we should just show it underneath the badge or progress bar. I will confirm with Anne on this as well.

dhartunian reviewed (changes requested):
@elkmaster This is a good start on the design follow-up but I think it needs some more work. Some of this is a mismatch between what I believe the design is and what has been implemented here. Let's chat tomorrow about this and make sure we're on the same page. I've included some commentary inline.

*I will be pushing up a commit on top to demonstrate the changes I'd like to see in the interest of time since we're about to hit stability period for the release.*

Continue your work on top of this and let me know what you think. Primarily, I'd like to create a clear way for the code to show how a job in a particular state is mapped to its visual representation. Right now this logic is scattered across many files and it's unclear whether it works correctly.

In addition there are a few visual things for you to follow-up on:
1. Badge widths are different in firefox vs chrome
![Screen Shot 2020-02-29 at 11 58 36 AM](https://user-images.githubusercontent.com/986307/75646259-08364900-5c17-11ea-85e8-8892ad0304be.png)


2. The dropdown selectors on firefox have some odd visual artifacts

![Screen Shot 2020-02-28 at 4 05 06 PM](https://user-images.githubusercontent.com/986307/75646271-11271a80-5c17-11ea-8d21-2c1281a1b702.png)

3. Job Details page does not show badges. The design includes them. Can you find a way to use the exact same components with different styling? That would be ideal. I would be willing to negotiate a change in the designs in order to make the components identical on the table and the job details page. Let's discuss with the team.

dhartunian committed:
ui: refactor and clean up job status visuals

Renamed `FractionCompleted` to `JobStatus` to clarify that it was
responsible for showing the entire status of a job since it includes the Badge.

Added function called `jobToVisual` that maps a job to a visual enum via its
status. `JobStatus` uses it to decide waht to show. The goal is to make the
`JobStatus` component map the enums to different visual components but
not be responsible for deciding which one to choose. Same change in
`JobDetails`

Exported `BadgeStatus` from badge so we can refine the
`jobStatusToBadgeStatus`function.

Removed `renamedStatuses` since the name didn't help us understand what
the purpose of the function was.

`Progress` component renamed to `ProgressBar` and no longer responsible
for showing the badge.

vladlos committed:
ui: fix and refactor job status

Changed `Badge` width  to `max-content` for fix it on firefox
Used `Pagination` component
Removed artifacts for dropdown selectors on firefox
Used `DATE_FORMAT` constant
refactor and clean up code

Release note (ui): none

nathanstilwell commented on pkg/ui/src/views/shared/components/dropdown/index.tsx:
```diff
@@ -71,12 +71,7 @@ export default class Dropdown extends React.Component<DropdownOwnProps, {}> {
-    }
-    return <span className="active"><Icon type="caret-down" /></span>;
-  }
+  arrowRenderer = () => <span className="active"><Icon type="caret-up" /></span>;
```
Does this only ever render an "up caret" now?

elkmaster replied:
yes, that was an actual request. but after it will be redesigned again. see https://github.com/cockroachdb/cockroach/pull/45802

nathanstilwell replied:
👍

nathanstilwell reviewed (commented):
Only a few comments. In general I agree with David and Andrii's comments.

vladlos committed:
ui: job details status

Used `Job Status Cell` component for Job details

Release note (ui): none

dhartunian reviewed (approved):
Beautiful @elkmaster! Thx for fixing all the visual nits and cleaning up the code. Feels much better.

dhartunian reviewed (changes requested):
Sorry found a few more issues we'll need to resolve before merging.

1. Since we've removed the "top 50"/"all" dropdown, we currently only retrieve the last 50 jobs. This is bad.
2. When have a multi-page set of jobs and I'm on, say, page 3, if I stay there for a bit the UI reverts back to page 1. Something is hooked into a data refresh that doesn't preserve the page I'm on.

I'll figure out what to do with this @elkmaster, no need to push changes at the moment.

dhartunian committed:
ui: add back show options dropdown to jobs page

In order to retain compatibility with our existing API, we need to retain
this dropdown to allow the user to load all jobs if they so choose.

Because pagination is not implemented completely on the backend,
we're retaining this ability to let customers deal with potentially large
job lists if they choose.

Release note: None

dhartunian committed:
ui: fix job page revert to page 1 bug

When the jobs page was sitting at a page > 1, it would periodically
refresh itself back to page 1. This was due to the fact that the
logic deciding whether the new data in the table was "different"
was incorrect in multiple ways.

It compared the entire cached jobs object which contained
cache-specific information that was not relevant here.

Release note: none

dhartunian reviewed (approved):
I addressed the outstanding issues by adding back the dropdown for 50/all and fixing the refresh bug.
